Source/WebCore/rendering/RenderGrid.h

```diff
@@ -46,6 +46,16 @@ struct ContentAlignmentData {
 
 enum class GridAxisPosition : uint8_t { GridAxisStart, GridAxisEnd, GridAxisCenter };
 
+class GridItemSizeCache {
+public:
+    void setSizeForGridItem(const RenderBox& gridItem, LayoutUnit size);
+    std::optional<LayoutUnit> sizeForItem(const RenderBox& gridItem) const;
+    void invalidateSizeForItem(const RenderBox& gridItem);
+
+private:
+    SingleThreadWeakHashMap<const RenderBox, std::optional<LayoutUnit>> m_sizes;
```

Comment:
Why don't we want to enforce a grid item to always have a size if we are caching it?

Reply:
The reason I put a `std::optional<LayoutUnit>` for the value is so that we could use `std::nullopt` as the empty value in the HashMap. We could always override that with a specific value for the empty value but I'm not really sure what, in theory, could be "correct." 

That's also why I designed the cache's API in a way such that this is invisible to the user in it. To them a cached item will always have a value since that's the only way to add things to the cache